tests/adv_diff/bp_adv_diff_01.cpp

```diff
@@ -56,33 +56,29 @@ evaluate_brinkman_bc_callback_fcn(int B_idx,
 #endif
 
     Pointer<PatchHierarchy<NDIM> > patch_hierarchy = hier_math_ops->getPatchHierarchy();
-    const int coarsest_ln = 0;
     const int finest_ln = patch_hierarchy->getFinestLevelNumber();
 
-    for (int ln = coarsest_ln; ln <= finest_ln; ++ln)
+    Pointer<PatchLevel<NDIM> > finest_level = patch_hierarchy->getPatchLevel(finest_ln);
+    for (PatchLevel<NDIM>::Iterator p(finest_level); p; p++)
     {
-        Pointer<PatchLevel<NDIM> > level = patch_hierarchy->getPatchLevel(ln);
-        for (PatchLevel<NDIM>::Iterator p(level); p; p++)
-        {
-            Pointer<Patch<NDIM> > patch = level->getPatch(p());
-            const Box<NDIM>& patch_box = patch->getBox();
-            Pointer<SideData<NDIM, double> > B_data = patch->getPatchData(B_idx);
+        Pointer<Patch<NDIM> > patch = finest_level->getPatch(p());
+        const Box<NDIM>& patch_box = patch->getBox();
+        Pointer<SideData<NDIM, double> > B_data = patch->getPatchData(B_idx);
 
-            for (unsigned int axis = 0; axis < NDIM; ++axis)
+        for (unsigned int axis = 0; axis < NDIM; ++axis)
+        {
+            for (Box<NDIM>::Iterator it(SideGeometry<NDIM>::toSideBox(patch_box, axis)); it; it++)
             {
-                for (Box<NDIM>::Iterator it(SideGeometry<NDIM>::toSideBox(patch_box, axis)); it; it++)
-                {
-                    SideIndex<NDIM> si(it(), axis, SideIndex<NDIM>::Lower);
-                    IBTK::Vector coord = IndexUtilities::getSideCenter(*patch, si);
+                SideIndex<NDIM> si(it(), axis, SideIndex<NDIM>::Lower);
+                IBTK::Vector coord = IndexUtilities::getSideCenter(*patch, si);
```

Comment:
This can also lead to processor dependent results since side center is computed from patch lower. Perhaps you can add a version in IndexUtilities class where you send grid geometry pointer to the routine to compute side coordinate from grid_lower_x.

Reply:
I added routine in IndexUtilities to compute side-center using grid_lower_X. Please take a look. Now all side-center calculations in Brinkman examples and tests are done using this new routine.